pkcs11/pkcs11.go

```diff
@@ -1687,3 +1706,169 @@ func (s *Slot) generateECDSA(o keyOptions) (crypto.PrivateKey, error) {
 	}
 	return priv, nil
 }
+
+func WithHash(privKey crypto.PrivateKey, hash crypto.Hash) (crypto.PrivateKey, error) {
```

Comment:
Determine the hash based on the crypto.DecrypterOptions passed to the decrypter.

For OAEP that's: https://pkg.go.dev/crypto/rsa#OAEPOptions

It looks like PKCS #1 v1.5 decryption doesn't use a hash: https://pkg.go.dev/crypto/rsa#PKCS1v15DecryptOptions